dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# ============================
+# Observation Planning Toolbox
+# ============================
+"""
+The observation planning toolbox consists of a number of entities that
+simplify the planning and scheduling of observations by abstracting much
+of the boilerplate code necessary to accomplish the same thing using
+core astropy or ephemerides objects.  These can be thought of as common
+idioms or convenience classes and functions that would be recreated over
+and over by astronomers and or software engineers building telescope
+scheduling software.
+
+Important assumptions:
+
+- It should be convenient to specify dates and times in the local timezone
+  of the observer.
+
+- It should be convenient to do timezone conversions easily.
+
+  This means methods that can take dates can either take astropy
+  dates (as used internally) or python datetime objects (not naive, but
+  tagged with timezone).  So dates can be passed in as timezone-aware
+  datetime objects and results can be easily converted between such.
+
+"""
+
+# ================
+# Observer objects
+# ================
+"""
+An observer defines a single terrestrial location and applicable
+environmental attributes thereof that affect observation calculations
+at that location.
+"""
+# Define an observer by full specification
+#
+# Because there may be many attributes for an observer, I suggest using
+# keyword parameters for most of them.
+from astroplan.entity import Observer
+
+obs = Observer(name='Subaru Telescope',
+               longitude='-155:28:48.900',
+               latitude='+19:49:42.600',
+               elevation=4163,
```

Comment:
As per my comment above, I think it's ok in this instance to allow *both* cases.  Definitely we need to have the *option* of what @adrn suggests here, because there are some ways of creating an `EarthLocation` that are more complex that simply giving `lat` `lon` and `elevation` (i.e., creating from x/y/z or from other coordinate objects).  But it also makes sense that in a lot of standard cases you just want to give a single set of keywords and leave it at that.

However, I'm 100% in agreement with @adrn that `elevation`, `pressure`, and `temperature` need to be quantities: they are unitful numbers, so the user should have to specify the unit - and anyway, they'll be stored with their units as part of `AltAz`, so you'll need to provide the unit somehow anyway.  For lat/lon it's fine to allow strings: in fact, all you need to say is "latitude and longitude can be anything that `EarthLocation` accepts", which maps directly onto the `Latitude`/`Longitude` classes, just as the comment a few lines down mentions.